tests/test_db.py

```diff
@@ -0,0 +1,118 @@
+        park_id = await db.insert_parking_lot(parking_lot)
+        assert park_id == 1
+
+        async with db.pool.acquire() as conn:
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 1
+
+
+@pytest.mark.asyncio
+async def test_delete_parking_lot(event_loop):
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
+        await db.insert_parking_lot(parking_lot)
+        async with db.pool.acquire() as conn:
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 1
+            park_id = await db.delete_parking_lot(1)
+            assert park_id == 1
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 0
+
+
+@pytest.mark.asyncio
+async def test_delete_parking_lot_unsuccessful(event_loop):
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
+        await db.insert_parking_lot(parking_lot)
+        async with db.pool.acquire() as conn:
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 1
+            park_id = await db.delete_parking_lot(2)
+            assert park_id is None
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 1
+
+
+@pytest.mark.asyncio
+async def test_update_parking_lot_price_and_availability(event_loop):
+    new_availability = 10
+    new_price = 5
+
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
+        park_id = await db.insert_parking_lot(parking_lot)
+        park_id = await db.update_parking_lot_availability(park_id, new_availability)
+        assert park_id == 1
+        park_id = await db.update_parking_lot_price(park_id, new_price)
+        assert park_id == 1
+
+        async with db.pool.acquire() as conn:
+            p_record: Record = await conn.fetchrow('SELECT * from ParkingLots;')
+
+        assert p_record['num_available'] == new_availability
+        assert p_record['price'] == new_price
+
+
+@pytest.mark.asyncio
+async def test_update_parking_lot_price_and_availability_unsuccessful(event_loop):
+    new_availability = 10
+    new_price = 5
+
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
```

Comment:
Do we do any checking that we don't insert two parking lots into the same location?

If so, that should become a test, too.

Reply:
We don't do any checks like that as any new parking lots should be created by our 'admins' and shouldn't change that often. I'd move on from it now. Pls.

Reply:
Move on from what?

Reply:
As in let's not do it now, please. :) This PR already blocks Harvey and me from working on other stuff.

Reply:
If it's blocking you and seems to be working then merge it.

Reply:
@HarveyHunt merge when you've had another look. (does not need to be tonight).